rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java

```diff
@@ -60,6 +62,24 @@ private Object callEvalDynamicList(MethodContext context, Object[] objects) {
             }
         }
 
-        return  jedis.evalsha(context.getSha1(), keys, argv);
+        return  evalShaHandleReloadScript(context, keys, argv);
+    }
+
+    private Object evalShaHandleReloadScript(MethodContext context, List<String> keys, List<String> argv) {
+        try {
+            return jedis.evalsha(context.getSha1(), keys, argv);
+        } catch (JedisDataException e) {
+            if (e.getMessage().contains("NOSCRIPT")) {
+                //If it throws again, we can back-off or we can just let it throw again. In this case, I think we should
+                //let it throw because most likely will be trying the same thing again and hopefully it will succeed later.
+                final String sha2 = jedis.scriptLoad(context.getLuaContext().getRenderedLuaString());
```

Comment:
being that sha2 is a legitimate hashing algorithm, I would recommend choosing an alternate variable name.